pwndbg/abi.py

```diff
@@ -111,3 +116,31 @@ class SigreturnABI(SyscallABI):
 linux_i386_srop = ABI(['eax'], 4, 0)
 linux_amd64_srop = ABI(['rax'], 4, 0)
 linux_arm_srop = ABI(['r7'], 4, 0)
+
+
+# See issue #264 for more reference
+# 'GNU/Linux' for linux
+# 'none' for bare metal
+abi = None
+
+def is_linux():
+    return 'Linux' in abi
+
+@pwndbg.events.start
+def update():
+    global abi
+
+    # detect current ABI of client side by 'show osabi'
+    osabi_string = gdb.execute('show osabi', to_string=True)
+    match = re.search('currently "([^"]+)"', osabi_string)
+    if match:
+        abi = match.group(1)
+
+def Linux(func):
+    @functools.wraps(func)
+    def caller(*args, **kwargs):
+        # call func only if ABI is Linux
+        if is_linux():
+            func(*args, **kwargsa)
```

Comment:
missing `return` statement

Reply:
We might also want there to be a way to specify a default return value, e.g.

```py
@LinuxOnly("Not Linux!")
def get_some_string():
  return "Yes Linux!"
```

Which can be done by using a `class` for the wrapper instead of a function.

```py
class LinuxOnly(object):
  def __init__(self, default):
    self._default = default
  def __call__(self, function):
    def wrapper(*a,**kw):
      if not linux: return self._default
      else: return function(*a,**kw)
```